src/languages/ilanguagesservice.h

```diff
@@ -35,14 +36,18 @@ class ILanguagesService : MODULE_EXPORT_INTERFACE
 public:
     virtual ~ILanguagesService() = default;
 
-    virtual ValCh<LanguagesHash> languages() const = 0;
-    virtual ValCh<Language> currentLanguage() const = 0;
+    virtual const LanguagesHash& languages() const = 0;
+    virtual Language language(const QString& languageCode) const = 0;
+    virtual const Language& currentLanguage() const = 0;
+    virtual async::Notification currentLanguageChanged() const = 0;
 
-    virtual LanguageStatus::Status languageStatus(const QString& languageCode) const = 0;
+    virtual bool hasPlaceholderLanguage() const = 0;
+    virtual const Language& placeholderLanguage() const = 0;
```

Comment:
What is the purpose of "placeholder" language?

Reply:
It is the thing that @shoogle added here: #12553